### ISSUE TITLE
Add support for Text Selection Action Bar for Android in Godot 4.4

### ISSUE THREAD

**Issue by UnderGamer05** [godotengine/godot-proposals#11028]
### Describe the project you are working on

I'm working on a project to improve the usability of the Android Editor text fields in Godot 4.4. This enhancement focuses on adding a standard Android text selection Action Bar, which appears after text is highlighted. The Action Bar provides options like Cut, Copy, Paste, Select All, and potentially other context-specific actions. This feature will create a more intuitive, consistent, and user-friendly experience for Android developers and users interacting with text fields within the Godot Editor on Android devices.

### Describe the problem or limitation you are having in your project

Currently, the Android Editor in Godot does not support a native text selection Action Bar. When users select text within text fields, they cannot access typical Android text manipulation actions (e.g., Cut, Copy, Paste, etc.), resulting in a lack of parity with native Android apps. This gap disrupts the user experience and requires users to rely on less efficient, non-standard methods for managing text.

The lack of a native text selection Action Bar also limits accessibility and productivity on Android, as users cannot utilize familiar, built-in editing capabilities. This can be especially problematic in text-heavy applications like note-taking apps, messaging, or any application that requires frequent text manipulation.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

The proposed enhancement is to integrate the Android-native text selection Action Bar within the Godot Editor when running on Android. This feature will allow users to access Cut, Copy, Paste, and Select All options natively when they select text, creating a consistent Android experience.

The Action Bar will appear automatically when the user highlights text within a text field, offering standard Android text-editing capabilities. This feature aligns Godot's Android experience with user expectations for text management on Android devices, boosting usability and accessibility across text-heavy applications.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

The implementation will involve:
1. **Integration of Native Android Text Selection API:** Utilize the Android API for handling text selection and Action Bar options. Godot should invoke this API when text is selected in an editor field to present the Action Bar with Cut, Copy, Paste, and other options.
2. **Customizable Action Menu:** Offer the ability for developers to define additional options on the Action Bar if needed, in alignment with how other Android applications manage custom actions. This would allow adding specialized options (like "Translate" or "Share") to match the application's specific needs.
3. **Example Code and Implementation:** is a pseudo-code example of how this could be implemented:

```java
// Example Java code for adding Action Mode Callbacks in an Android text field
TextView textView = findViewById(R.id.text_editor);

// Configure the Action Mode Callbacks for text selection
textView.setCustomSelectionActionModeCallback(new ActionMode.Callback() {
    @Override
    public boolean onCreateActionMode(ActionMode mode, Menu menu) {
        // Inflate or add custom actions to the menu here
        mode.getMenuInflater().inflate(R.menu.text_selection_menu, menu);
        return true;
    }

    @Override
    public boolean onPrepareActionMode(ActionMode mode, Menu menu) {
        // If there’s any dynamic update required before displaying
        return false;
    }

    @Override
    public boolean onActionItemClicked(ActionMode mode, MenuItem item) {
        // Handle action clicks, e.g., Cut, Copy, Paste
        switch (item.getItemId()) {
            case android.R.id.cut:
                // Implement Cut functionality
                return true;
            case android.R.id.copy:
                // Implement Copy functionality
                return true;
            case android.R.id.paste:
                // Implement Paste functionality
                return true;
            default:
                return false;
        }
    }

    @Override
    public void onDestroyActionMode(ActionMode mode) {
        // Clean up any resources if needed
    }
});
```
• **Mock-up**: The Action Bar appears directly above the text field, displaying options like Cut, Copy, and Paste when text is selected.

4. **Godot Integration:** This could be handled through a `GodotActivity` extension on Android where the Action Bar would be activated for specific nodes like `TextEdit` and `LineEdit`. Custom hooks could be introduced to trigger the Action Mode and manage the Android Menu for various text manipulation actions.

Also see: 
https://developer.android.com/reference/android/view/ActionMode
https://developer.android.com/reference/android/widget/TextView#setCustomSelectionActionModeCallback(android.view.ActionMode.Callback)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

A script-based workaround to replicate the native Android Action Bar’s functionality is impractical. While it is possible to add basic text editing options through custom UI nodes, this approach lacks the native integration, fluidity, and accessibility provided by the Android Action Bar. Mimicking the exact look, feel, and behavior of Android's text handling would be difficult and would introduce significant complexity without the benefits of a true native experience.

### Is there a reason why this should be core and not an add-on in the asset library?

This feature should be core to Godot's Android platform support rather than an add-on, as it is a fundamental component of user interaction on Android devices. Including this natively would provide a seamless experience aligned with Android standards, eliminating the need for developers to implement custom Action Bars. Additionally, this enhancement will support accessibility compliance on Android, improving usability for all users and ensuring consistency in text manipulation behavior.

**Comment by syntaxerror247** [comment]
I don't think this is possible in Godot. Godot doesn't use Android UI elements, only way to have selection handle is to create one from scratch in Godot.

**Comment by UnderGamer05** [comment]
I understand, but in some other engine has a extra text fields which apear after selecting a text.

![Screenshot_20241027_105014_Prisma3D](https://github.com/user-attachments/assets/245abc32-af1f-4a50-8721-4960bcf5cbef)
![Screenshot_20241027_104915_Prisma3D](https://github.com/user-attachments/assets/f165e318-108a-447f-84b2-28a5c4d54850)
(Application name is Prisma3d)

Like above screenshot it can be done by adding a another text field which can handle text Selection Action and sync it with godot active text fields. 

or if you have any other option.

**Comment by syntaxerror247** [comment]
That is good for single line text, but this won't be a viable solution for Godot `Code Editor`.

This is a WIP addon [MobileCaret](https://github.com/EiTaNBaRiBoA/MobileCaret), you can try this, last time i tested, it was working prety good with `LineEdit`, it don't have context menu but that can be implemented using popupPanel.
